pandas/core/series.py

```diff
@@ -4054,7 +4054,21 @@ def _try_cast(arr, take_fast_path):
                                            isinstance(subarr, np.ndarray))):
                 subarr = construct_1d_object_array_from_listlike(subarr)
             elif not is_extension_type(subarr):
-                subarr = np.array(subarr, dtype=dtype, copy=copy)
+                subarr2 = np.array(subarr, dtype=dtype, copy=copy)
+
```

Comment:
do not put any more code here! these routines are already way overloaded.

also I believe you can simply do this by:

```
(dtype, arr) = cast.infer_dtype_from_array(subarr)
subarray = np.array(arr, dtype=dtype)
```

Reply:
then pls make a routine in cast, this is not the right place for this. we have so many casting routines  pls modify / use existing ones.

Reply:
> When using that approach wouldn't cast [1.0, 2.0, None] to ['1.0', '2.0', None].

That is actually what happened in 0.22:

```
In [30]: pd.Series([1.0, 2.0, None], dtype=str).values
Out[30]: array([1.0, 2.0, None], dtype=object)
```

(but the new behaviour to convert the numbers to strings is certainly an improvement)

Reply:
Ahh... OK hold off on merging then. Ideally this would just be a bugfix for that regression :/

Won't have time to look till ~3 hours from now.